From 3fd7acc6e5a8c316c366827956d7c06affdc9685 Mon Sep 17 00:00:00 2001
From: Olivier Dion <olivier.dion@savoirfairelinux.com>
Date: Fri, 7 Oct 2022 15:20:37 -0400
Subject: [PATCH] test/unitTest: Seperate pure tests from integration tests

So we can now do `make check` in a reproducible way.

To enable integration tests `./configure --enable-integration-tests`.

Change-Id: Iaa68c77abade9fe78bae881abc25e034bcb6a067
---
With file names adjusted for the source tarball.

 configure.ac              |  5 ++++
 test/unitTest/Makefile.am | 52 +++++++++++++++++++++------------------
 2 files changed, 33 insertions(+), 24 deletions(-)

diff --git a/configure.ac b/configure.ac
index 27215d044..aaeef1c1b 100644
--- a/daemon/configure.ac
+++ b/daemon/configure.ac
@@ -40,6 +40,11 @@ AS_IF([test "x$enable_debug" = "xyes"],
   [DAEMONCFLAGS+=" -DNDEBUG=1 -O3"
    DAEMONCXXFLAGS+=" -DNDEBUG=1 -O3 -Wno-deprecated"])
 
+AC_ARG_ENABLE([enable_integration_tests],
+  AS_HELP_STRING([--enable-integration-tests],
+    [Also run integration tests with check target]))
+AM_CONDITIONAL([ENABLE_INTEGRATION_TESTS], [test "x$enable_integration_tests" = "xyes"])
+
 AC_ARG_ENABLE([fuzzing],
   AS_HELP_STRING([--enable-fuzzing],
     [Build fuzzing tools]))
diff --git a/test/unitTest/Makefile.am b/test/unitTest/Makefile.am
index f2b0fd994..ab232ddb7 100644
--- a/daemon/test/unitTest/Makefile.am
+++ b/daemon/test/unitTest/Makefile.am
@@ -13,30 +13,6 @@ check_PROGRAMS =
 
 ####### Unit Test ########
 
-#
-# account_factory
-#
-check_PROGRAMS += ut_account_factory
-ut_account_factory_SOURCES = account_factory/testAccount_factory.cpp common.cpp
-
-#
-# account_archive
-#
-check_PROGRAMS += ut_account_archive
-ut_account_archive_SOURCES = account_archive/account_archive.cpp common.cpp
-
-#
-# migration
-#
-check_PROGRAMS += ut_migration
-ut_migration_SOURCES = account_archive/migration.cpp common.cpp
-
-#
-# certstore
-#
-check_PROGRAMS += ut_certstore
-ut_certstore_SOURCES = certstore.cpp common.cpp
-
 #
 # scheduler
 #
@@ -127,6 +103,32 @@ ut_video_scaler_SOURCES = media/video/test_video_scaler.cpp common.cpp
 check_PROGRAMS += ut_audio_frame_resizer
 ut_audio_frame_resizer_SOURCES = media/audio/test_audio_frame_resizer.cpp common.cpp
 
+if ENABLE_INTEGRATION_TESTS
+
+#
+# account_factory
+#
+check_PROGRAMS += ut_account_factory
+ut_account_factory_SOURCES = account_factory/testAccount_factory.cpp common.cpp
+
+#
+# account_archive
+#
+check_PROGRAMS += ut_account_archive
+ut_account_archive_SOURCES = account_archive/account_archive.cpp common.cpp
+
+#
+# migration
+#
+check_PROGRAMS += ut_migration
+ut_migration_SOURCES = account_archive/migration.cpp common.cpp
+
+#
+# certstore
+#
+check_PROGRAMS += ut_certstore
+ut_certstore_SOURCES = certstore.cpp common.cpp
+
 #
 # call
 #
@@ -237,4 +239,6 @@ ut_sip_srtp_SOURCES = sip_account/sip_srtp.cpp
 check_PROGRAMS += ut_plugins
 ut_plugins_SOURCES = plugins/plugins.cpp common.cpp
 
+endif # ENABLE_INTEGRATION_TESTS
+
 TESTS = $(check_PROGRAMS)
-- 
2.37.3

